hack/generator/pkg/astmodel/code_generation_context.go

```diff
@@ -23,17 +23,15 @@ type CodeGenerationContext struct {
 // New CodeGenerationContext creates a new immutable code generation context
 func NewCodeGenerationContext(
 	currentPackage PackageReference,
-	packageImports map[PackageImport]struct{},
+	packageImports *PackageImportSet,
 	generatedPackages map[PackageReference]*PackageDefinition) *CodeGenerationContext {
 
-	packageImportsMap := make(map[PackageReference]PackageImport)
-	for imp := range packageImports {
-		packageImportsMap[imp.PackageReference] = imp
-	}
+	imports := EmptyPackageImportSet()
```

Comment:
Wondering if this being `NewPackageImportSet` would:
1. Help discoverability through autocompletion (since `New`/`Make` are usually prefixes for these sorts of things?)
2. More closely match other "constructor" funcs.